pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -342,18 +354,21 @@ func (gc *GarbageCollector) attemptToDeleteWorker(ctx context.Context, item inte
 		return forgetItem
 	}
 
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithValues(logger, "item", n)
```

Comment:
I'm having second thoughts about this and other `WithValues` calls: before we add those, we should have some kind of before/after comparison of the performance impact.

@ncdc: can we perhaps merge this PR with just the conversion of the existing calls, then enhance the log content in a second PR? Then we can do the benchmarking for that one.

Reply:
@pohly I just pushed another commit that should undo the extra logged fields. Is that looking better (and did I miss anything)? Thanks!